core/src/main/java/org/renjin/primitives/combine/LazyBuilder.java

```diff
@@ -63,11 +63,11 @@ private Vector composeNames(String prefix, Vector vector) {
     Vector names = vector.getNames();
     int numElements = vector.length();
 
-    if(!Strings.isNullOrEmpty(prefix) || names != null) {
+    if(!Strings.isNullOrEmpty(prefix) || (names !=  Null.INSTANCE && names != null)) {
       hasNames = true;
     }
 
-    if(Strings.isNullOrEmpty(prefix) && names == null) {
+    if(Strings.isNullOrEmpty(prefix) && (names ==  Null.INSTANCE || names == null)) {
```

Comment:
Would you mind dropping the `names != null` and `names == null` checks alltogether? `null` should be avoided alltogether in Renjin (don't know why I relapsed here) and I think it's best to keep that standard clear. 

You could add an `@Nonnull` to vector.getNames() to remove any ambiguity.